### PR TITLE
Update VPCNetworkConfiguration CR based on NSX VPC changes

### PR DESCRIPTION
**Pull request by lxiaopei**
1. Add LoadBalancerVPCEndpointEnabled param to indicate whether support for load balancing is needed. Setting this flag to true causes allocation of private IPs from the private block associated with this VPC tobe used by the load balancer.
2. Change PublicIPv4Blocks to ExternalIPv4Blocks.

### CODE REVIEW

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go [hunk 1]
```diff
@@ -25,10 +25,13 @@ type VPCNetworkConfigurationSpec struct {
 	EdgeClusterPath string `json:"edgeClusterPath,omitempty"`
 	// NSX-T Project the Namespace associated with.
 	NSXTProject string `json:"nsxtProject,omitempty"`
-	// NSX-T IPv4 Block paths used to allocate public Subnets.
+	// Flag to indicate whether support for load balancing is needed
+	// +kubebuilder:default=false
+	LoadBalancerVPCEndpointEnabled bool `json:"loadBalancerVPCEndpointEnabled,omitempty"`
```
Comment (jianjuns) [inline pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:30]: VPC Endpoint is a term in AWS with a different meaning, so let us change a name unless NSX-T VPC uses it.

How about CreateLoadBalancerSubnet if it is just about creating a Subnet for AVI SE?

Comment (jianjuns) [reply pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:30]: And we should keep the T1 networking behavior to create the Subnet using the reserved CIDR for SEs, not allocating from VPC Private CIDR. Please update the commit message for that.

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go [hunk 2]
```diff
@@ -13,6 +13,12 @@ const (
 	AccessModePrivate string = "private"
 )
 
+// Field to indicate whether support for load balancing is needed
```
Comment (jianjuns) [inline pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:16]: Nit: add a "." to the end of the sentence.

**Review by jianjuns — DISMISSED** [review]

**Review by seanpang-vmware — DISMISSED** [review]
LGTM

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go (hunk 1 above)
Comment (lxiaopei) [reply pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go]: in NSX VPC API, term "LoadBalancerVPCEndpoint" is used in VPC payload, we keep consistent with NSX VPC API.

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go (hunk 2 above)
Comment (lxiaopei) [reply pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:16]: done, thanks.

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go [hunk 3]
```diff
@@ -25,10 +31,12 @@ type VPCNetworkConfigurationSpec struct {
 	EdgeClusterPath string `json:"edgeClusterPath,omitempty"`
 	// NSX-T Project the Namespace associated with.
 	NSXTProject string `json:"nsxtProject,omitempty"`
-	// NSX-T IPv4 Block paths used to allocate public Subnets.
+	// Flag to indicate whether support for load balancing is needed
```
Comment (jianjuns) [inline pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:34]: Add ending "."

Comment (jianjuns) [reply pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:34]: It is no longer a "flag".

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go [hunk 4]
```diff
@@ -13,6 +13,12 @@ const (
 	AccessModePrivate string = "private"
 )
 
+// Field to indicate whether support for load balancing is needed.
```
Comment (jianjuns) [inline pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:16]: Say: "Load balancer endpoint configuration"?

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go (hunk 1 above)
Comment (lxiaopei) [reply pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go]: Confirmed with Avi team that the AVI SE subnet will be created from VPC Private IPBlock. If LoadBalancerVPCEndpoint enabled set to true, when creating VPC, a Private subnet will be auto-created for Avi SE.

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go (hunk 3 above)
Comment (lxiaopei) [reply pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:34]: thanks Jianjun, changed.

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go (hunk 4 above)
Comment (lxiaopei) [reply pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:16]: Sounds good, changed.

**Review by jianjuns — APPROVED** [review]

**Pull request merged**